staging/src/k8s.io/sample-controller/hack/update-codegen.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SCRIPT_ROOT=$(dirname ${BASH_SOURCE})/..
+CODEGEN_PKG=${CODEGEN_PKG:-$(cd ${SCRIPT_ROOT}; ls -d -1 ./vendor/k8s.io/code-generator 2>/dev/null || echo ../code-generator)}
+source ${CODEGEN_PKG}/hack/lib/codegen.sh
+
+codegen::generate-groups "deepcopy,client,informer,lister" "$(dirname ${BASH_SOURCE})/../../.." k8s.io/sample-controller/pkg/client k8s.io/sample-controller/pkg/apis example:v1alpha1 --go-header-file ${SCRIPT_ROOT}/hack/custom-boilerplate.go.txt
```

Comment:
the output-base `"$(dirname ${BASH_SOURCE})/../../.."` shouldn't be necessary anymore

Reply:
I commented `--go-header-file ${SCRIPT_ROOT}/hack/custom-boilerplate.go.txt` out in sample-apiserver because otherwise the global kube code-gen scripts override our generated files with the default boilerplate. Would like to have a way to exclude repos from the global scripts.

Reply:
Updated to reflect both these comments.

The `update-codegen.sh` script now looks similar to the one in `sample-apiserver`

Reply:
How come the global scripts overwrite it? `./hack/update-codegen.sh` (in the root of the repo) simply calls out to the `update-codegen.sh` script in each staging repo. Unless you mean a different global script?

Reply:
Makefile.generated_files.